File: packages/android-client/FishjamClient/src/main/java/com/fishjamdev/client/Command.kt
package com.fishjamdev.client

import kotlinx.coroutines.CompletableJob
import kotlinx.coroutines.Job

internal enum class CommandName {
  CONNECT,
  JOIN,
  ADD_TRACK,
  REMOVE_TRACK,
  RENEGOTIATE,
  LEAVE
}

internal enum class ClientState {
  CREATED,
  CONNECTED,
  JOINED
}

internal data class Command(
  val commandName: CommandName,
  val clientStateAfterCommand: ClientState?,
  val command: () -> Unit = {}
) {
  constructor(commandName: CommandName, command: () -> Unit = {}) : this(commandName, null, command)

  val job: CompletableJob = Job()
}

File: packages/ios-client/Sources/MembraneRTC/Command.swift
import Foundation

enum CommandName {
    case connect
    case join
    case addTrack
    case removeTrack
    case renegotiate
    case leave
}

enum ClientState {
    case created
    case connected
    case joined
}

class Command {
    let commandName: CommandName
    let clientStateAfterCommand: ClientState?
    let execute: () -> Void

    var job = DispatchGroup()

    init(commandName: CommandName, clientStateAfterCommand: ClientState? = nil, execute: @escaping () -> Void = {}) {
        self.commandName = commandName
        self.clientStateAfterCommand = clientStateAfterCommand
        self.execute = execute
    }
}
